Minos/Minos.Site/Controllers/AdminController.cs

```diff
@@ -272,5 +272,33 @@ public IActionResult DeletarPergunta(int id)
             return RedirectToAction("CadastrarPergunta", "Admin");
         }
 
+        [HttpGet]
+        public IActionResult AtualizarPergunta(int id)
+        {
+            ViewBag.Pergunta = _perguntaRepository.ObterPerguntaPeloId(id);
+            return View();
+        }
+
+        [HttpPost]
+        public IActionResult AtualizarPergunta(string texto, int id)
+        {
+
+            Pergunta pergunta = new Pergunta { Texto = texto, Id = id };
```

Comment:
@RobsonJunior1994 não faz sentido você criar um objeto pergunta e só se for válido, buscá-lo no banco. Você deve ir no banco obtê-lo primeiro e depois atualizar suas informações em memória. E aí sim, verificar se está válido após suas mudanças. Se estiver, manda atualizar pelo Repository.

Reply:
Do jeito atual eu verifico primeiramente se o usuário enviou uma pergunta válida, se ele não enviou eu não gasto nem tempo atualizando as informações em memoria e buscando no banco de dados.

Do jeito que você falou para fazer:

            Pergunta pergunta = null;
            pergunta = _perguntaRepository.ObterPerguntaPeloId(id);
            pergunta.Texto = texto;
            if (pergunta.EhValida())
            {
                _perguntaRepository.Atualizar(pergunta);
            }

da um bug na view  dando a entender que consegui salvar uma pergunta com espaços, porem ele cai no else e não altera nada no banco, mas a view fica preenchida com espaços.

Reply:
@lucasnobredev me ajudou a entender o porque de acontecer esse erro na view, ao tentar atualizar uma pergunta somente com espaços ele não salvava no banco mas, ficava preso a memoria a associação com espaços da propriedade texto de pergunta por conta da injeção de dependências com singleton, o @lucasnobredev idenficou o problema e alterou para AddScoped que atualizado tova vez que se faz um request.